test/unit/filters/filterDeadlinePassedSpec.js

```diff
@@ -35,7 +35,7 @@ describe('Test Deadline Passed filter', function () {
 
             someDate.setDate(someDate.getDate() + numberOfDaysToAdd);
 
-            var dd = ("0" + (someDate.getDate() + 1)).slice(-2),
+            var dd = ("0" + someDate.getDate() ).slice(-2),
                 mm = ("0" + (someDate.getMonth() + 1)).slice(-2),
                 y = someDate.getFullYear();
             var date = y + '-' + mm + '-' + dd;
```

Comment:
What about removing all this complicated logic that deals with dates and start using [`moment.js`](https://momentjs.com/) ? Then we can just do `moment().add(5, 'days')`.

Reply:
Yes, I don't see why this change fixes the problem so the code is difficult to understand.

Since we don't currently need `moment.js` besides on this test, you can add it as DevDependency (https://github.com/inspirehep/inspirehep-search-js/blob/master/package.json#L22) and then use it in this test.